mlir/include/mlir/Interfaces/ValueBoundsOpInterface.h

```diff
@@ -265,10 +265,28 @@ class ValueBoundsConstraintSet {
 
   ValueBoundsConstraintSet(MLIRContext *ctx);
 
+  /// A callback to allow injecting custom value bounds constraints.
+  /// It takes the current value, the dim (or kIndexValue), and a reference to
+  /// the constraints set.
+  using PopulateCustomValueBoundsFn =
```

Comment:
I'm wondering if `PopulateCustomValueBoundsFn` is actually needed. We could make `ValueBoundsConstraintSet::addBound` public.